### PR TITLE
Default DENY all egress from pods in the namespace

### PR DESCRIPTION
**Pull request by knabben**
**What type of PR is this?**
/kind feature
/area test
/sig network

**What this PR does / why we need it**:
Add a DENY default-deny-egress to pods in the namespace

**Which issue(s) this PR fixes**:

Refs https://github.com/kubernetes/kubernetes/issues/97640

**Special notes for your reviewer**:
12 - DENY default-deny-egress to a NS

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/network/netpol/network_policy.go [hunk 1]
```diff
@@ -586,6 +586,19 @@ var _ = SIGDescribeCopy("Netpol [LinuxOnly]", func() {
 			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachabilityIsolated})
 		})
 
+		ginkgo.It("should deny egress all pods in a namespace [Feature:NetworkPolicy] ", func() {
```
Comment (mattfenwick) [inline test/e2e/network/netpol/network_policy.go:589]: missing word `should deny egress *from* all pods in a namespace`

File: test/e2e/network/netpol/network_policy.go [hunk 2]
```diff
@@ -586,6 +586,19 @@ var _ = SIGDescribeCopy("Netpol [LinuxOnly]", func() {
...
+			nsX, _, _, model, k8s := getK8SModel(f)
+			policy := GetDenyEgress("deny-egress-pod-a")
```
Comment (mattfenwick) [inline test/e2e/network/netpol/network_policy.go:591]: why `pod-a` ?

File: test/e2e/network/netpol/policies.go [hunk 3]
```diff
@@ -26,19 +26,34 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// GetDenyIngress returns a default deny policy named 'name'.
+// GetDenyIngress returns a default deny ingress policy.
```
[the remark this answers is not in the document]
Comment (knabben) [reply test/e2e/network/netpol/policies.go:29]: I change in propose, `the policy named 'name'` looks a little bit awkward, but I can hold back here.

**Comment by mattfenwick** [comment]
/lgtm 

will let somebody else approve this!

**Comment by mattfenwick** [comment]
/lgtm

assuming the tests pass

🎉 🎉

**Pull request merged**